pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -117,16 +123,25 @@ const (
 )
 
 func (c *Cluster) makeJob(osdProps osdProperties, provisionConfig *provisionConfig) (*batch.Job, error) {
+	osdOnPVC := osdProps.pvc.ClaimName != ""
+	osdOnPVCMetadata := osdProps.metadataPVC.ClaimName != ""
 
 	podSpec, err := c.provisionPodTemplateSpec(osdProps, v1.RestartPolicyOnFailure, provisionConfig)
 	if err != nil {
 		return nil, err
 	}
 
-	if osdProps.pvc.ClaimName == "" {
+	if !osdOnPVC {
 		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: osdProps.crushHostname}
 	} else {
+		// This is not needed in raw mode and 14.2.8 brings it
+		// but we still want to do this not to loose backward compatibility with lvm based OSDs...
 		podSpec.Spec.InitContainers = append(podSpec.Spec.InitContainers, c.getPVCInitContainer(osdProps.pvc))
+		if osdOnPVCMetadata {
+			// FIX ME
+			// change /srv to something else?
```

Comment:
why is it using /srv or what should it be?

Reply:
Well we need to put the block somewhere, `/mnt` is already used and we can't move to subdir since existing cluster have that. So I put it in `/srv`. Ideally, we would have had `/mnt/data/` and `/mnt/metadata/`.